src/PubSub/Queue.php

```diff
@@ -0,0 +1,274 @@
+     */
+    public function pop($queue = null)
+    {
+        $topic = $this->getTopic($queue);
+
+        $subscription = $this->subscribeToTopic($topic);
+
+        $messages = $subscription->pull([
+            'returnImmediately' => true,
+            'maxMessages'       => 1,
+        ]);
+        if (!empty($messages) && count($messages) > 0) {
+            $class = $this->getClass($queue);
+
+            return new Job(
+                $this->container,
+                $this,
+                $messages[0],
+                $class,
+                $this->connectionName,
+                $queue
+            );
+        }
+    }
+
+    /**
+     * Push an array of jobs onto the queue.
+     *
+     * @param  array  $jobs
+     * @param  mixed  $data
+     * @param  string $queue
+     * @return mixed
+     */
+    public function bulk($jobs, $data = '', $queue = null)
+    {
+        $payloads = [];
+        foreach ((array) $jobs as $job) {
+            $payloads[] = ['data' => $this->createPayload($job, $queue, $data)];
+        }
+        $topic = $this->getTopic($queue);
+        $this->subscribeToTopic($topic);
+
+        return $topic->publishBatch($payloads);
+    }
+
+    /**
+     * Acknowledge a message.
+     *
+     * @param  \Google\Cloud\PubSub\Message $message
+     * @param  string                       $queue
+     */
+    public function acknowledge(Message $message, $queue = null)
+    {
+        $topic        = $this->getTopic($queue);
+        $subscription = $topic->subscription($this->getSubscriberName($topic));
+        $subscription->acknowledge($message);
+    }
+
+    /**
+     * Acknowledge a message and republish it onto the queue.
+     *
+     * @param  \Google\Cloud\PubSub\Message $message
+     * @param  string                       $queue
+     * @return mixed
+     */
+    public function acknowledgeAndPublish(Message $message, $queue = null, $options = null, $delay = 0)
+    {
+        $topic        = $this->getTopic($queue);
+        $subscription = $topic->subscription($this->getSubscriberName($topic));
+        $subscription->acknowledge($message);
+        $options = array_merge([
+            'available_at' => (string) $this->availableAt($delay),
+        ], $options);
+
+        return $topic->publish([
+            'data'       => $message->data(),
+            'attributes' => $options,
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createPayload($job, $queue, $data = '')
+    {
+        if (!$job instanceof DispatcherJob) {
+            return base64_encode(parent::createPayload($job, $data, $queue));
+        }
+
+        return base64_encode(json_encode($job->getPayload()));
+    }
+
+    /**
+     * Get the current topic
+     *
+     * @param $queue
+     * @return Topic
+     */
+    public function getTopic($queue)
+    {
+        $queue = $queue ?: $this->default;
+        $topic = $this->pubsub->topic($queue);
```

Comment:
`return ($this->pubsub->topic($queue));` ?

AKA why initializing a var and adding memory to the heap only to return it?